TODO: these tests need mock adjustments.
=========================== short test summary info ============================
FAILED tests/unit/test_client.py::Test__create_gapic_client::test_w_emulator
FAILED tests/unit/test_client.py::TestClient::test_constructor_defaults - Ass...
FAILED tests/unit/test_client.py::TestClient::test_constructor_with_emulator_host
3 failed, 1241 passed, 10 warnings in 28.75s

--- a/google/cloud/bigtable/client.py
+++ b/google/cloud/bigtable/client.py
@@ -67,14 +67,15 @@ DATA_SCOPE = "https://www.googleapis.com/auth/bigtable.data"
 READ_ONLY_SCOPE = "https://www.googleapis.com/auth/bigtable.data.readonly"
 """Scope for reading table data."""
 
+
 def _create_gapic_client(client_class, client_options=None, transport=None):
     def inner(self):
         return client_class(
-                credentials=None,
-                client_info=self._client_info,
-                client_options=client_options,
-                transport=transport,
-            )
+            credentials=None,
+            client_info=self._client_info,
+            client_options=client_options,
+            transport=transport,
+        )
 
     return inner
 
@@ -206,11 +207,15 @@ class Client(ClientWithProject):
         # Note: this code also exists in the firestore client.
         if "GrpcAsyncIOTransport" in str(transport.__name__):
             return grpc.aio.secure_channel(
-                self._emulator_host, self._local_composite_credentials(), options=options
+                self._emulator_host,
+                self._local_composite_credentials(),
+                options=options,
             )
         else:
             return grpc.secure_channel(
-                self._emulator_host, self._local_composite_credentials(), options=options
+                self._emulator_host,
+                self._local_composite_credentials(),
+                options=options,
             )
 
     def _local_composite_credentials(self):
@@ -241,11 +246,11 @@ class Client(ClientWithProject):
 
     def _create_gapic_client_channel(self, client_class, grpc_transport):
         options = {
-                    "grpc.max_send_message_length": -1,
-                    "grpc.max_receive_message_length": -1,
-                    "grpc.keepalive_time_ms": 30000,
-                    "grpc.keepalive_timeout_ms": 10000,
-                }.items()
+            "grpc.max_send_message_length": -1,
+            "grpc.max_receive_message_length": -1,
+            "grpc.keepalive_time_ms": 30000,
+            "grpc.keepalive_timeout_ms": 10000,
+        }.items()
         if self._client_options and self._client_options.api_endpoint:
             api_endpoint = self._client_options.api_endpoint
         else:
@@ -257,9 +262,7 @@ class Client(ClientWithProject):
             channel = self._emulator_channel(grpc_transport, options)
         else:
             channel = grpc_transport.create_channel(
-                host=api_endpoint,
-                credentials=self._credentials,
-                options=options,
+                host=api_endpoint, credentials=self._credentials, options=options,
             )
         transport = grpc_transport(channel=channel, host=api_endpoint)
         return transport

--- a/noxfile.py
+++ b/noxfile.py
@@ -33,6 +33,7 @@ UNIT_TEST_PYTHON_VERSIONS = ["3.6", "3.7", "3.8", "3.9"]
 # 'docfx' is excluded since it only needs to run in 'docs-presubmit'
 nox.options.sessions = [
     "unit",
+    "system_emulated",
     "system",
     "cover",
     "lint",
@@ -110,12 +111,37 @@ def unit(session):
     """Run the unit test suite."""
     default(session)
 
+@nox.session(python="3.8")
+def system_emulated(session):
+    import subprocess
+    
+    try:
+        subprocess.call(["gcloud", '--version'])
+    except OSError:
+        session.skip("gcloud not found but required for emulator support")
+
+    hostport = "localhost:8789"
+    p = subprocess.Popen([
+        "gcloud", "beta", "emulators", "bigtable", "start",
+        "--host-port", hostport
+    ])
+
+    system(session, emulator=hostport)
+
+    # Stop Emulator
+    p.send_signal(1)
 
 @nox.session(python=SYSTEM_TEST_PYTHON_VERSIONS)
-def system(session):
+def system(session, emulator=False):
     """Run the system test suite."""
     system_test_path = os.path.join("tests", "system.py")
     system_test_folder_path = os.path.join("tests", "system")
+
+    if emulator:
+        os.environ["BIGTABLE_EMULATOR_HOST"] = emulator
+    elif os.environ.get("BIGTABLE_EMULATOR_HOST"):
+        del os.environ["BIGTABLE_EMULATOR_HOST"]
+
 
     # Check the value of `RUN_SYSTEM_TESTS` env var. It defaults to true.
     if os.environ.get("RUN_SYSTEM_TESTS", "true") == "false":

--- a/noxfile.py
+++ b/noxfile.py
@@ -114,6 +114,7 @@ def unit(session):
 @nox.session(python="3.8")
 def system_emulated(session):
     import subprocess
+    import signal
     
     try:
         subprocess.call(["gcloud", '--version'])
@@ -126,22 +127,17 @@ def system_emulated(session):
         "--host-port", hostport
     ])
 
-    system(session, emulator=hostport)
+    session.env["BIGTABLE_EMULATOR_HOST"] = hostport
+    system(session)
 
     # Stop Emulator
-    p.send_signal(1)
+    os.killpg(os.getpgid(p.pid), signal.SIGTERM)
 
 @nox.session(python=SYSTEM_TEST_PYTHON_VERSIONS)
-def system(session, emulator=False):
+def system(session):
     """Run the system test suite."""
     system_test_path = os.path.join("tests", "system.py")
     system_test_folder_path = os.path.join("tests", "system")
-
-    if emulator:
-        os.environ["BIGTABLE_EMULATOR_HOST"] = emulator
-    elif os.environ.get("BIGTABLE_EMULATOR_HOST"):
-        del os.environ["BIGTABLE_EMULATOR_HOST"]
-
 
     # Check the value of `RUN_SYSTEM_TESTS` env var. It defaults to true.
     if os.environ.get("RUN_SYSTEM_TESTS", "true") == "false":

--- a/tests/unit/test_client.py
+++ b/tests/unit/test_client.py
@@ -67,16 +67,24 @@ class Test__create_gapic_client(unittest.TestCase):
         client_class = mock.Mock()
         emulator_host = emulator_channel = object()
         credentials = _make_credentials()
+
+        client_options = mock.Mock()
+        transport = mock.Mock()
+
         client = _Client(
             credentials, emulator_host=emulator_host, emulator_channel=emulator_channel
         )
         client_info = client._client_info = mock.Mock()
-
-        result = self._invoke_client_factory(client_class)(client)
+        result = self._invoke_client_factory(
+            client_class, client_options=client_options, transport=transport
+        )(client)
 
         self.assertIs(result, client_class.return_value)
         client_class.assert_called_once_with(
-            channel=client._emulator_channel, client_info=client_info
+            credentials=None,
+            client_info=client_info,
+            client_options=client_options,
+            transport=transport,
         )
 
 
@@ -121,7 +129,6 @@ class TestClient(unittest.TestCase):
         self.assertIs(client._client_info, _CLIENT_INFO)
         self.assertIsNone(client._channel)
         self.assertIsNone(client._emulator_host)
-        self.assertIsNone(client._emulator_channel)
         self.assertEqual(client.SCOPE, (DATA_SCOPE,))
 
     def test_constructor_explicit(self):
@@ -167,22 +174,23 @@ class TestClient(unittest.TestCase):
 
         credentials = _make_credentials()
         emulator_host = "localhost:8081"
-        with mock.patch("os.getenv") as getenv:
-            getenv.return_value = emulator_host
-            with mock.patch("grpc.insecure_channel") as factory:
-                getenv.return_value = emulator_host
+        with mock.patch("os.environ", {BIGTABLE_EMULATOR: emulator_host}):
+            with mock.patch("grpc.secure_channel") as factory:
                 client = self._make_one(project=self.PROJECT, credentials=credentials)
+                # don't test local_composite_credentials
+                client._local_composite_credentials = lambda: credentials
+                # channels are formed when needed, so access a client
+                # create a gapic channel
+                client.table_data_client
 
         self.assertEqual(client._emulator_host, emulator_host)
-        self.assertIs(client._emulator_channel, factory.return_value)
-        factory.assert_called_once_with(
-            target=emulator_host,
-            options={
-                "grpc.keepalive_time_ms": 30000,
-                "grpc.keepalive_timeout_ms": 10000,
-            }.items(),
-        )
-        getenv.assert_called_once_with(BIGTABLE_EMULATOR)
+        options = {
+            "grpc.max_send_message_length": -1,
+            "grpc.max_receive_message_length": -1,
+            "grpc.keepalive_time_ms": 30000,
+            "grpc.keepalive_timeout_ms": 10000,
+        }.items()
+        factory.assert_called_once_with(emulator_host, credentials, options=options)
 
     def test__get_scopes_default(self):
         from google.cloud.bigtable.client import DATA_SCOPE

